.travis.yml

```diff
@@ -18,9 +18,17 @@ before_install:
             sudo docker pull welder/web:latest
             sudo docker pull welder/web-with-coverage:latest
             sudo docker pull welder/web-e2e-tests:latest
-        else
+        fi
+install:
+  - |
+        if [ "$COMMAND" != "end-to-end-test" ]; then
             npm install
         fi
+        if [ "$COMMAND" == "cockpit-test" ]; then
+            node run build --with-coverage
+            sed -i "s,var welderApiHost=.*,var welderApiHost=\"api\";," ./public/js/config.js
```

Comment:
FYI I don't like this very much but I don't have another proposal ATM. It bugs me that for e2e testing we build the app inside the docker image and for testing with Cockpit we build the app outside and just mount it under the appropriate directory. However this can be resolved some other time.